test/e2e/backup-resource/deletion.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package backupresource
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+
+	. "github.com/vmware-tanzu/velero/test/e2e"
+	. "github.com/vmware-tanzu/velero/test/e2e/util/k8s"
+	. "github.com/vmware-tanzu/velero/test/e2e/util/kibishii"
+	. "github.com/vmware-tanzu/velero/test/e2e/util/providers"
+	. "github.com/vmware-tanzu/velero/test/e2e/util/velero"
+)
+
+const (
+	deletionTest = "deletion-workload"
+)
+
+// Test backup and restore of Kibishi using restic
+
+func Backup_deletion_with_snapshots() {
+	backup_deletion_test(true)
+}
+
+func Backup_deletion_with_restic() {
+	backup_deletion_test(false)
+}
+func backup_deletion_test(useVolumeSnapshots bool) {
```

Comment:
Does this case check the snapshots or restic data? If it doesn't, why we need to back up the volumes?

Reply:
Snapshot verification will be provide in a new PR.